### ISSUE TITLE
Please enable support for Istio on macOS ARM64

### ISSUE THREAD

**Issue by WesleyMConner** [canonical/microk8s#3168]
#### Summary
Istio appears to support macOS AMD64 and macOS ARM64:
- https://github.com/istio/istio/releases/download/1.13.4/istioctl-1.13.4-osx.tar.gz
- https://github.com/istio/istio/releases/download/1.13.4/istioctl-1.13.4-osx-arm64.tar.gz

MicroK8s appears to have enabled Istio for macOS AMD64, but not Istio for macOS ARM64:
- `microk8s enable istio` available on macOS AMD64
- `microk8s enable istio` returns exit code 1 on macOS ARM64 

#### Why is this important?
Istio adoption on macOS is pending a consistent offering on AMD64 and ARM64 platforms.

#### Are you interested in contributing to this feature?
- Was preparing to test/exercise macOS AMD64 and ARM64 platforms side-by-side.
- Currently reviewing options - i.e., Linkerd, MicroK8s alternatives

**Comment by ktsakalozos** [comment]
Hi @WesleyMConner 

The scripts called when you `microk8s enable` and `microk8s disable` istio are in [1] and they are less than 40 lines each. Would you be interested in submitting a patch to add ARM64 support to them?

[1] https://github.com/canonical/microk8s-community-addons/tree/main/addons/istio

**Comment by WesleyMConner** [comment]
@ktsakalozos - Thanks for the quick response and link.

Tactically, I fell back on Linkerd since it seemed to _ready to roll_ on both architectures.

At a glance, the script looks pretty straightforward. I'll try to find a little time next week to extend/test on AMD64/ARM64 in parallel.

**Comment by WesleyMConner** [comment]
Just a quick follow up.

I took an initial peek this afternoon at enabling istio on arm64.

Trivial changes were required to two files:
* `/var/snap/microk8s/common/addons/community/addons/istio/enable`
* `/var/snap/microk8s/common/addons/community/addons.yaml`

I saw that the istio entry  in `/var/snap/microk8s/common/addons/community/tests/validators.py` was hard-wired to fail for all cases but x86_64. I tried commenting out the forced failure to see if existing AMD64 tests would cover ARM64 as well. Unfortunately, I don't think my changes to validators.py were getting picked up.

The good news is that tests on my AMD64 macOS box indicated no harm ...
- Installation and verification tests were both fine.

On my ARM64 macOS box ...
- `/snap/bin/microk8s.istioctl verify-install` still produces "Istio is not available for arm64".

So, there's more for me to look at  next week ... as time permits.
